File: examples/go/example_cli.go
package main

import (
	"encoding/json"
	"flag"
	"fmt"
	"os"
)

type output struct {
	Ok      bool `json:"ok"`
	Version int  `json:"version"`
	Seed    int  `json:"seed"`
	Data    struct {
		Hello string `json:"hello"`
	} `json:"data"`
}

func main() {
	jsonFlag := flag.Bool("json", false, "emit JSON to stdout")
	_ = flag.Bool("stable", false, "no-op; output already deterministic via struct fields")
	seed := flag.Int("seed", 0, "seed for deterministic behavior")
	flag.Parse()

	if !*jsonFlag {
		fmt.Fprintln(os.Stderr, "example_cli.go: missing --json")
		os.Exit(2)
	}

	fmt.Fprintf(os.Stderr, "example_cli.go: seed=%d\n", *seed)

	var out output
	out.Ok = true
	out.Version = 1
	out.Seed = *seed
	out.Data.Hello = "world"

	b, err := json.MarshalIndent(out, "", "  ")
	if err != nil {
		fmt.Fprintf(os.Stderr, "example_cli.go: json marshal failed: %v\n", err)
		os.Exit(1)
	}
	fmt.Println(string(b))
}

File: examples/node/example_cli.js
#!/usr/bin/env node

function main() {
  const args = process.argv.slice(2);

  const has = (flag) => args.includes(flag);
  const argValue = (flag, def) => {
    const i = args.indexOf(flag);
    if (i === -1) return def;
    return args[i + 1] ?? def;
  };

  if (!has("--json")) {
    console.error("example_cli.js: missing --json");
    process.exit(2);
  }

  const seed = Number(argValue("--seed", "0"));
  console.error(`example_cli.js: seed=${seed}`);

  const out = {
    ok: true,
    version: 1,
    seed,
    data: { hello: "world" },
  };

  process.stdout.write(JSON.stringify(out, null, 2) + "\n");
}

main();
